tests/lib/prepare-restore.sh

```diff
@@ -621,12 +618,6 @@ prepare_suite_each() {
             ausearch -i -m AVC --checkpoint "$RUNTIME_STATE_PATH/audit-stamp" || true
             ;;
     esac
-
-    # Check for invariants late, in order to detect any bugs in the code above.
-    if [[ "$variant" = full ]]; then
-        "$TESTSTOOLS"/cleanup-state pre-invariant
-    fi
-    tests.invariant check
```

Comment:
Would it make sense to keep this check too (in case prepare does something wrong) or is it useless after these changes?

Reply:
done, sorry, forgot to push that part

Reply:
@sergiocazzolato are you sure you pushed the change, I still don't see `tests.invariant check` being run in the `prepare_suite_each` function in this file

Reply:
I agree here. I think it's ok to move the cleanup to restore, but keep the invariant check both here and in restore. The check does not look too expensive compared to actual tests, and by keeping it in both places we make sure that we start with a good system, and nothing slips through the cracks between restore and prepare.

Reply:
Thanks for the review, so now the invariant check is done as part of the prepare and restore of each test. It is a quick check so it should not affect on performance.